storage/s3_sdk/list_gcs_buckets_test.go

```diff
@@ -17,37 +17,83 @@ package s3sdk
 
 import (
 	"bytes"
-	"io/ioutil"
+	"context"
+	"fmt"
 	"log"
 	"os"
 	"strings"
 	"testing"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
 )
 
-func TestMain(m *testing.M) {
-	log.SetOutput(ioutil.Discard)
-	s := m.Run()
-	log.SetOutput(os.Stderr)
-	os.Exit(s)
-}
 
 func TestList(t *testing.T) {
-	t.Skip("HMAC secrets are not set up correctly: https://github.com/GoogleCloudPlatform/golang-samples/issues/1213")
-	googleAccessKeyID := os.Getenv("STORAGE_HMAC_ACCESS_KEY_ID")
-	googleAccessKeySecret := os.Getenv("STORAGE_HMAC_ACCESS_SECRET_KEY")
+	ctx := context.Background()
+	tc := testutil.SystemTest(t)
 
-	if googleAccessKeyID == "" || googleAccessKeySecret == "" {
-		t.Skip("STORAGE_HMAC_ACCESS_KEY_ID and STORAGE_HMAC_ACCESS_SECRET_KEY must be set. Skipping.")
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("storage.NewClient: %v", err)
 	}
+	defer client.Close()
 
-	buf := new(bytes.Buffer)
-	_, err := listGCSBuckets(buf, googleAccessKeyID, googleAccessKeySecret)
+	// Set up service account HMAC key to use for this test.
+	key, err := createTestKey(ctx, client, tc.ProjectID)
 	if err != nil {
-		t.Errorf("listGCSBuckets: %v", err)
+		log.Fatalf("error setting up HMAC key: %v", err)
+	}
+	defer deleteTestKey(ctx, client, key)
+
+	log.Printf("ProjectID: %v", tc.ProjectID)
+	log.Printf("key: %v", key.AccessID)
+	log.Printf("service account: %v", key.ServiceAccountEmail)
+
+	var buf *bytes.Buffer
+	// New HMAC key may take up to 15s to propagate.
+	timeout := time.After(15 * time.Second)
```

Comment:
Please use `testutil.Retry`. Be sure to update the failure to `r.Errorf` instead of calling `t.Fatalf` directly.

Reply:
Done, thanks for noting this.